spring-data-jpa/src/main/java/org/springframework/data/jpa/repository/query/ParameterBinding.java

```diff
@@ -182,6 +188,115 @@ public boolean isCompatibleWith(ParameterBinding other) {
 		return other.getClass() == getClass() && other.getOrigin().equals(getOrigin());
 	}
 
+	/**
+	 * Represents a {@link ParameterBinding} in a JPQL query augmented with instructions of how to apply a parameter as an
+	 * {@code IN} parameter.
+	 *
+	 * @author Thomas Darimont
+	 * @author Mark Paluch
+	 */
+	static class PartTreeParameterBinding extends ParameterBinding {
+
+		private final Class<?> parameterType;
+		private final JpqlQueryTemplates templates;
+		private final EscapeCharacter escape;
+		private final Type type;
+		private final boolean ignoreCase;
+		private final boolean noWildcards;
+
+		public PartTreeParameterBinding(BindingIdentifier identifier, ParameterOrigin origin, Class<?> parameterType,
+				Part part, @Nullable Object value, JpqlQueryTemplates templates, EscapeCharacter escape) {
+
+			super(identifier, origin);
+
+			this.parameterType = parameterType;
+			this.templates = templates;
+			this.escape = escape;
+
+			this.type = value == null && Type.SIMPLE_PROPERTY.equals(part.getType()) ? Type.IS_NULL : part.getType();
+			this.ignoreCase = Part.IgnoreCaseType.ALWAYS.equals(part.shouldIgnoreCase());
+			this.noWildcards = part.getProperty().getLeafProperty().isCollection();
+		}
+
+		/**
+		 * Returns whether the parameter shall be considered an {@literal IS NULL} parameter.
+		 */
+		public boolean isIsNullParameter() {
+			return Type.IS_NULL.equals(type);
+		}
+
+		@Override
+		public Object prepare(@Nullable Object value) {
+
+			if (value == null || parameterType == null) {
+				return value;
+			}
+
+			if (String.class.equals(parameterType) && !noWildcards) {
+
+				switch (type) {
+					case STARTING_WITH:
+						return String.format("%s%%", escape.escape(value.toString()));
+					case ENDING_WITH:
+						return String.format("%%%s", escape.escape(value.toString()));
+					case CONTAINING:
+					case NOT_CONTAINING:
+						return String.format("%%%s%%", escape.escape(value.toString()));
+					default:
+						return value;
+				}
```

Comment:
```suggestion
                              String escapedValue = escape.escape(value.toString());
                              return switch (type) {
                                  case STARTING_WITH -> escapedValue + "%";
                                  case ENDING_WITH -> "%" + escapedValue;
                                  case CONTAINING, NOT_CONTAINING -> "%" + escapedValue + "%";
                                  default -> value;
                              };
```
Consider leveraging the `switch` expression for its readability and declaring `escapedValue` once to avoid redundancy.